lib/metadata/metadataUtils.js

```diff
@@ -26,7 +26,13 @@ function getNullVersion(objMD, bucketName, objectKey, log, cb) {
         // the latest version is not the null version, but null version exists
         log.debug('null version exists, get the null version');
         options.versionId = objMD.nullVersionId;
-        return metadata.getObjectMD(bucketName, objectKey, options, log, cb);
+        return metadata.getObjectMD(bucketName, objectKey, options, log, (err, nullVersionMD) => {
+            if (err && err.is.NoSuchKey) {
```

Comment:
Not sure it its possible for it to be a non-Arsenal error, but better safe than sorry.
```suggestion
            if (err && err.is?.NoSuchKey) {
```